logs_ingest/filtering.py

```diff
@@ -0,0 +1,117 @@
+    RESOURCE_ID_ATTRIBUTE
+
+GLOBAL = "GLOBAL"
+AMOUNT_OF_DOT_SEPARATORS_IN_GLOBAL_FILTERS = 2
+
+
+class LogFilter:
+    def __init__(self):
+        self._filter_config: str = os.environ.get("FILTER_CONFIG", "")
+        filter_config_pattern = re.compile(r'([^;\s].+?)=([^;]*)')
+        self._filters_tuples = filter_config_pattern.findall(self._filter_config)
+        self._filters_tuples = [self._prepare_filters_tuples(filter_tuple) for filter_tuple in self._filters_tuples]
+        self.filters_dict = self._prepare_filters_dict()
+        if self.filters_dict:
+            logging.info(f"Applied filter_config: {self._filter_config}")
+
+    @staticmethod
+    def _prepare_filters_tuples(filter_tuple):
+        filter_name = filter_tuple[0].strip()
+        value = filter_tuple[1].strip()
+        if filter_name.count(".") > AMOUNT_OF_DOT_SEPARATORS_IN_GLOBAL_FILTERS:
+            key = ".".join(filter_name.split(".")[AMOUNT_OF_DOT_SEPARATORS_IN_GLOBAL_FILTERS + 1:])
+        else:
+            key = GLOBAL
+        return key, (filter_name, value)
+
+    def _prepare_filters_dict(self) -> Dict:
+        grouped_filters = self._group_filters()
+        filters_to_apply_dict = {}
+        for k, filter_name_value_dict in grouped_filters.items():
+            filters_to_apply = []
+            for filter_name, filter_value in filter_name_value_dict.items():
+                if "MIN_LOG_LEVEL" in filter_name:
+                    log_levels = self._get_log_levels(filter_value)
+                    if log_levels:
+                        log_level_filter = self._create_log_level_filter(log_levels)
+                        filters_to_apply.append(log_level_filter)
+                if "CONTAINS_PATTERN" in filter_name:
+                    contains_pattern_filter = self._create_contains_pattern_filter(filter_value)
+                    filters_to_apply.append(contains_pattern_filter)
+            filters_to_apply_dict[k] = filters_to_apply
+        return filters_to_apply_dict
+
+    def _group_filters(self) -> Dict:
+        filters_dict = {}
+        for key, filter_name_value in self._filters_tuples:
+            filters_dict.setdefault(key, {}).update({filter_name_value[0]: filter_name_value[1]})
+        return filters_dict
+
+    @staticmethod
+    def _create_log_level_filter(log_levels: Set):
+        return lambda severity, record: severity in log_levels
+
+    @staticmethod
+    def _create_contains_pattern_filter(pattern: str):
+        return lambda severity, record: fnmatch.fnmatch(record, pattern)
+
+    def should_filter_out_record(self, record: Dict, parsed_record: Dict) -> bool:
+        if not self.filters_dict:
+            return False
+
+        severity = parsed_record.get("severity", "")
+        resource_id = parsed_record.get(RESOURCE_ID_ATTRIBUTE, "")
+        resource_type = parsed_record.get(RESOURCE_TYPE_ATTRIBUTE, "")
+        record_as_string = json.dumps(record)
```

Comment:
Really not sure about that, but I thought we would match contains with parsed `content` - if we match full record, possibly we could have a lots of false positives with record metadata: Let's say my application prints out word `category` in log lines that I want to push to Dynatrace - with current implementation I'm unable to select them, as `CONTAINS_PATTERN` would pass all records, as they all have `category` metadata field.

Reply:
You're right, I should use `content ` for CONTAINS_PATTERN filter, I'll change that.